x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/ClientTransformIndexer.java

```diff
@@ -363,20 +376,23 @@ private void doSaveState(TransformState state, ActionListener<Void> listener) {
                             + statsExc.getMessage()
                     );
                     assert false : "[" + getJobId() + "] updating stats of transform failed, unexpected version conflict of internal state";
-                    // the version must be reset, otherwise it will never succeed for this indexer
-                    this.seqNoPrimaryTermAndIndex.set(null);
+
+                    // if we don't fix it, the transform will fail forever
+                    // get the stored doc to set seqNoPrimaryTermAndIndex correctly
```

Comment:
I am still not sure about the "fail forever" thing. 

Everything that updates the document goes through these methods on this class. It may not be the same thread, but it is the same state right?

So, if a previous state update occurred, (and usurped what we initially had in our seq_no and term), the atomic ref would be updated by line 326. 

I thought this conflict only occurred when there were two updates in flight. Both referencing the same seq_no and term, but one made it through, the other still referenced the old one (which got updated automatically when the successful call completed).